gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/TransfersAreUniqueValidatorTest.java

```diff
@@ -0,0 +1,119 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.notice.TransfersAreUniqueNotice;
+import org.mobilitydata.gtfsvalidator.table.GtfsTransfer;
+import org.mobilitydata.gtfsvalidator.table.GtfsTransferTableContainer;
+
+@RunWith(JUnit4.class)
+public class TransfersAreUniqueValidatorTest {
+  private TransfersAreUniqueValidator validator = new TransfersAreUniqueValidator();
+  private final GtfsTransfer transfer_1 = new GtfsTransfer.Builder()
+                                              .setFromStopId("stop_1")
+                                              .setToStopId("stop_2")
+                                              .setCsvRowNumber(1)
+                                              .build();
+  private final GtfsTransfer transfer_1_duplicate = new GtfsTransfer.Builder()
+                                                        .setFromStopId("stop_1")
+                                                        .setToStopId("stop_2")
+                                                        .setCsvRowNumber(10)
+                                                        .build();
+  private final GtfsTransfer transfer_2 = new GtfsTransfer.Builder()
+                                              .setFromStopId("stop_3")
+                                              .setToStopId("stop_7")
+                                              .setCsvRowNumber(2)
+                                              .build();
+  private final GtfsTransfer transfer_2_duplicate = new GtfsTransfer.Builder()
+                                                        .setFromStopId("stop_3")
+                                                        .setToStopId("stop_7")
+                                                        .setCsvRowNumber(4)
+                                                        .build();
+  private final GtfsTransfer transfer_3 = new GtfsTransfer.Builder()
+                                              .setFromStopId("stop_5")
+                                              .setToStopId("stop_9")
+                                              .setCsvRowNumber(3)
+                                              .build();
+
+  @Test
+  public void allDifferentTransfersShouldNotGenerateNotice() {
+    final NoticeContainer noticeContainer = new NoticeContainer();
+    List<GtfsTransfer> transfers =
```

Comment:
Is it necessary to convert the list into an ArrayList? Maybe give [this](https://www.baeldung.com/java-arrays-aslist-vs-new-arraylist) a read.

Can an immutable list be used?

Reply:
Changed `new ArrayList<>(Arrays.asList())` to `Arrays.asList()`. When you said 'can an immutable list be used?' do you mean immutable as in cannot be structurally modified i.e. Arrays.asList() or do you want it fully immutable? I.e. using `Collections.unmodifiableList()` or guava's `ImmutableList<>()`?

Reply:
I was suggesting using ImmutableList.of, but Arrays.asList is sufficient.

Reply:
Yeah, I was thinking [the Guava one](https://guava.dev/releases/snapshot/api/docs/com/google/common/collect/ImmutableList.html), i.e.

```
List<GtfsTransfer> transfers = ImmutableList.of(transfer_1, transfer_2, transfer_3);
```

I'm not sure how the `GtfsTransferTableContainer` treats the transfers so it might not work, but if it does it is probably better as immutable

Reply:
Ok, i will check if it works with an ImmutableList.

Reply:
Yep, ImmutableList works. I've changed all the `Arrays.asList` to `ImmutableList.of`.